pkg/subctl/cmd/validate_connections.go

```diff
@@ -0,0 +1,86 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+	submv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+)
+
+var validateConnectionsCmd = &cobra.Command{
+	Use:   "connections",
+	Short: "Validate the gateways connections",
+	Long: "This command validates if the connections of the gateways are all " +
+		"established",
+	Run: validateConnections,
+}
+
+func init() {
+	validateCmd.AddCommand(validateConnectionsCmd)
+}
+
+func validateConnections(cmd *cobra.Command, args []string) {
+	configs, err := getMultipleRestConfigs(kubeConfig, kubeContext)
+	exitOnError("Error getting REST config for cluster", err)
+
+	for _, item := range configs {
+		message := fmt.Sprintf("Validating Connections in %q", item.clusterName)
+		status.Start(message)
+		fmt.Println()
+		submariner := getSubmarinerResource(item.config)
```

Comment:
One thing that worries me about this is that if the submariner-operator is failing or stopped, the gateways will not be up-to date, and the data we provide will be out-dated or erroneous.

I still think we should stop grabbing the gateways from the submariner resources, and instead by Gateways.List()
 
@skitt  ^ 

Sorry to keep insisting in this, I'm known to be stubborn sometimes :-P , but I have faced many times where te operator is failing and it seems like it's not connected, but it is... or it seems like it's disconnected but it's not, we will have more complex escalations because of this.

Reply:
I agree, same as for metrics it seems preferable to go to the source (and I’ve probably changed my mind on this, ISTR advocating for going through the `Submariner` resource in the past).